### PR TITLE
[Dev 451] Auto-deployment Gh actions

### PR DESCRIPTION
**Pull request by dsh-otter**
not sure how to locally test it..

### CODE REVIEW

File: .github/workflows/main-auto-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+
+      - run: yarn
+      - run: yarn codegen
+      - run: yarn prepare:optimism
+      - run: yarn prepare:polygon
+      - run: yarn build:optimism && yarn deploy:optimism
+      - run: yarn build:polygon && yarn deploy:polygon
```
Comment (rahulrumalla) [inline .github/workflows/main-auto-deploy.yml:27]: We're missing mainnet here I blv

**Comment by rahulrumalla** [comment]
While we’re at it, can we make sure the build passes and tests run fine on PR, please? 

Maybe run build for Sepolia as part of the PR GH Action

File: .github/workflows/main-auto-deploy.yml (hunk 1 above)
Comment (dsh-otter) [reply .github/workflows/main-auto-deploy.yml:27]: ok, i didnt see a config for it in the code

Comment (rahulrumalla) [reply .github/workflows/main-auto-deploy.yml:27]: oh weird! I think we missed it - damn. Here's the SG link for it
https://thegraph.com/hosted-service/subgraph/otterspace-xyz/badges-mainnet

**Pull request merged**